core/backend/src/IModelDb.ts

```diff
@@ -1405,7 +1407,7 @@ export abstract class IModelDb extends IModel {
    * @note All saved `SettingDictionary`s are loaded into [[workspace.settings]] every time an iModel is opened.
    * @beta
    */
-  public saveSettingDictionary(name: string, dict: SettingDictionary) {
+  public saveSettingDictionary(name: string, dict: SettingObject) {
```

Comment:
Shouldn't `saveSettingDictionary` try to obtain the schema (file) lock?

Reply:
Yes, i suppose that's true. Changing the setting dictionary should definitely be an administrator task.